Entities.js

```diff
@@ -0,0 +1,339 @@
+    let randomNum = Math.random()*3
+
+    this.velocityX = randomNum;
+    this.velocityY = 3 - randomNum;
+    this.backgroundColor = 'red';
+    }
+  
+  update() {
+    // change velocity to move towards knight
+    this.velocityX += (GlobalX - this.x)/5000
+    this.velocityY += (GlobalY - this.y)/5000
+    //move
+    this.x += this.velocityX
+    this.y += this.velocityY
+
+    //check for intersection with walls
+    if(this.x >= (MAXwidth - this.width)){
+      this.reverse()
+    }
+    if(this.x <= 0){
+      this.reverse()
+    }
+    if(this.y >= (MAXheight - this.height)){
+      this.reverse()
+    }
+    if(this.y <= 0){
+      this.reverse()
+    }
+    
+//check for intersection with knight
+//check if there is a possible for up down
+//check if there is a possible for right left
+//check if those two things overlap
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
+    }
+    else if(this.x > GlobalX + playerWidth || this.x + this.width < GlobalX ){
+
+    }
+    else{
+      DeleteEntity(this.id)
+      counter += 1
+    }
+
+}
+reverse() {
+  this.velocityX = (this.velocityX * -1)
+  this.velocityY = (this.velocityY * -1)
+}
+}
+  
+
+class Grenade {
+  constructor(id) {
+    this.id = id
+    this.borderRadius = 75;
+    let x = 5
+    let y = Math.random()*500
+    this.x = x;
+    this.y = y;
+    this.height = 75;
+    this.width = 75;
+    this.renderer = <CreateCircle/>
+
+    let randomNum = Math.random()*3
+
+    this.velocityX = randomNum;
+    this.velocityY = 3 - randomNum;
+    this.backgroundColor = 'red';
+  }
+
+
+  update() {
+
+    //Bouncing off walls:
+    if(this.x >= (MAXwidth - this.width)){
+      this.reverse()
+    }
+    if(this.x <= 0){
+      this.reverse()
+    }
+    if(this.y >= (MAXheight - this.height)){
+      this.reverse()
+    }
+    if(this.y <= 0){
+      this.reverse()
+    }
+
+    //check for intersections with player
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
+    }
+    else if(this.x > GlobalX + playerWidth || this.x + this.width < GlobalX ){
+
+    }
+    else{
+      DeleteEntity(this.id)
+      counter += 1
+    }
+
+//updating the Grenades position
+    this.x += this.velocityX;
+    this.y += this.velocityY;
+  }
+
+  stop(){
+    this.velocityX = 0
+    this.velocityY = 0
+  }
+
+  reverse() {
+    this.velocityX = (this.velocityX * -1)
+    this.velocityY = (this.velocityY * -1)
+  }
+}
+
+class Shrapnel{
+  constructor(x,y,dX,dY,id){
+    this.id = id
+    this.borderRadius = 25;
+    this.x = x;
+    this.y = y;
+    this.height = 25;
+    this.width = 25;
+    this.velocityX = dX;
+    this.velocityY = dY;
+    this.renderer = <CreateCircle/>
+    this.backgroundColor = 'red';
+  }
+
+  update(){
+    this.x += this.velocityX
+    this.y += this.velocityY
+    this.velocityX += this.velocityX/75
+    this.velocityY += this.velocityY/75
+
+    //preliminary checks to see if the x or y by itself rules out the
```

Comment:
Cool idea, although it probably doesn't do that much in this case (and technically it could hurt, since now every time there *isn't* a collision you're adding an extra step)

For general advice: Don't worry too much about optimization at first. Use a tool that will help identify the slowest parts of your code (normally: which functions do I spend the most time inside, so not just slow functions, but commonly called ones matter most). Then go and optimize that part. Repeat if necessary.

Reply:
These are normally called "profilers"